### ISSUE TITLE
Enable JSON-based graph construction

### ISSUE THREAD

**Issue by nikhilk** [nikhilk/node-tensorflow#22]
We should be able to construct a graph in node.js using protocol-buffers directly. For example:

```javascript
var node1 = {
  name: 'Const',
  op: 'Const',
  attr: {
    dtype: {
      type: 3
    },
    value: {
      tensor: {
        dtype: 3,
        tensorShape: {},
        int_val: [1]
      }
    }
  }
}

var graph = {
  node: [
    node1
  ]
}

let graph = tf.Graph.fromGraphDefObject(graph)
```

Which uses the `encode` method of the compiled protobuf message.

While the encode and subsequent construction of a Graph and Session works fine, session.run results in a segmentation fault, suggesting an invalid attribute or some other detail of the graph.

**Comment by nikhilk** [comment]
Addressed in ecd28d71fa944ec7ecda920ee07650d9c371288c

**Comment by nikhilk** [comment]
Functionality included in 0.7.0.
